probs/probs.go

```diff
@@ -244,3 +245,12 @@ func CAA(detail string) *ProblemDetails {
 		HTTPStatus: http.StatusForbidden,
 	}
 }
+
+// DNS returns a ProblemDetails representing a DNSProblem
+func DNS(detail string) *ProblemDetails {
+	return &ProblemDetails{
+		Type:       DNSProblem,
+		Detail:     detail,
+		HTTPStatus: http.StatusForbidden,
```

Comment:
The places that are calling `probs.DNS` all seem to have previously been calling `probs.ConnectionFailure`. That function returns a prob with `HTTPStatus: http.StatusBadRequest`. I think `http.StatusForbidden` like used by `probs.DNS` is probably the more accurate choice but I wanted to confirm this change was intentional.

Reply:
Thanks for checking! It was not intentional (I copied it from CAA), but I agree in retrospect it seems better.

Reply:
So I agree `http.StatusBadRequest` is the wrong code but I also don't think `http.StatusForbidden` is either. We are not refusing to fulfill the request we literally _cannot_ fulfill the request.

The only times `probs.DNS` is now called is when DNS lookups fail (so get SERVFAIL or the actual exchange completely fails) which seems like an internal error and not necessarily something the user has any control over.

Reply:
> `probs.DNS` can get called on a variety of conditions: mainly SERVFAIL, NXDOMAIN, or timeout. However, we generally consider these all to be the user's "fault" in some sense since it generally means their DNS is broken.

So I don't think this ever gets called for NXDOMAIN, mainly because the various DNS `LookupXXX` methods never treat NXDOMAIN as an error, the lack of records is left up to the caller to deal with. That said I agree for things like SERVFAIL, it _probably_ was the users fault, and for timeouts it might be (or our resolver might just be slow or broken or something), ¯\_(ツ)_/¯.

> Also, this problem type is returned in a couple of cases: During validation, when the status doesn't matter, and during new-cert (recheckCAA). In the latter, the status also doesn't matter, because the problem types from the various domains get glommed into one. So the status code doesn't really matter; should I leave it out?

For now that is probably fine, although it opens up a bit of danger if someone down the road decides to use this in a place where it is needed.

Generally I think I've actually come back around that `http.StatusBadRequest` would be the right status to use if it were to get piped to the user (given this is what we use for i.e. a users server responding incorrectly/doing an invalid handshake etc).